### DISCUSSION TITLE
What's the best way to determine if SageTV is alive or not?

### DISCUSSION THREAD

**Discussion by wayner9** [OpenSageTV/Community#101] [General]
Occasionally my SageTV will stop working.  What is the best way to run a cron job to query whether SageTV is alive or not?  See if you can get a response from a web page for the SageTV Web UI?  Or is there a better way.

**Comment by gdippel** [comment]
You could try installing Sage tray (if you have Windows.) I use it to monitor a Sage Docker installation on UnRaid from my Windows PC.

**Reply by wayner9** [reply]
Do you know where to find it?  Sagetray sounds vaguely familiar - what I am looking for is something to run in a cron job on about an hourly basis.  And if the SageTV server is down it will send me an email or other notification to let me fix it.

It may just use the SageX API which is pretty much the same as the web UI - so just seeing if you get a valid response from the Sage web interface may effectively be the same thing.

**Comment by nyplayer** [comment]
A simple curl command should return an errorlevel of 0 ...
curl "http://sage:frey@127.0.0.1:8080/sage"

**Reply by wayner9** [reply]
Fair enough, but that is the same as I proposed initially but technically that is querying to see if the web UI is alive, not directly the SageTV engine.  I have seen instances where the web UI dies but SageTV server is still working.

**Reply by nyplayer** [reply]
I run my Server on Windows and do not use the Service I use SageTV running minimized on system tray.... so I use this below to check if SageTV is running if not start it... Checks very 5 minutes ... not sure about running a cron job to check SageTV status in Linux.

```
if not exist c:\temp md c:\temp
:loop
tasklist > c:\temp\ChkSageTV.txt
find "SageTV" c:\temp\ChkSageTV.txt
set taskerror=%ERRORLEVEL%
if %taskerror% NEQ 0 Goto StartSage
timeout 300
goto loop
:StartSage
cd "C:\Program Files\SageTV\SageTV"
c:
start SageTV -startup
goto loop
```

**Comment by sic0048** [comment]
Do you actually have a problem with SageTV not running?  

I will occasionally have a problem where my tuners disappear.  That's because something has changed and SageTV no longer "sees" the SageDTC tuners I have set up.  That requires that I stop the SageTV service, restart the SageDTC service, and then start the SageTV service back up after a 10-15 second wait. 

However the SageTV service is still running (and running correctly without any errors) when my tuners are "missing".

**Reply by wayner9** [reply]
Very occasionally I will have a crash on one of my unRAID servers or part of the SageTV process stops working - like the webUI.  I just like to know about this proactively and be able to deal with it.  I have an autistic daughter who watches SageTV and she gets very upset when it isn't working.

I have now set up a cron job that tries to access the SageTV web UI.  If that returns an error then it sends me an email  I am getting a decent amount of false negatives, which could be for other reasons like my internet is down for a bit in the middle of the night due to maintenance by my ISP.  Right now that cron job runs on my LAN but I just got a small VPS on Digital Ocean for a few dollars per month.   I am moving this process to the VPS so that will also let me know if my home internet is down.

**Comment by Swagger200** [comment]
I have all my servers running a web server (plug-in created by Nielm). If Sage crashes the web pages won't load. Can also set up recordings there incorporating Schedules Direct program guides.

**Comment by Swagger200** [comment]
![SageWebServer](https://github.com/user-attachments/assets/560eaefb-0efa-4cb0-9d98-2708c95f41da)

**Comment by Swagger200** [comment]
![SageWebServer2](https://github.com/user-attachments/assets/b9ceab1a-ba1d-4342-9aa1-62fcf69a0914)
